usim/_primitives/task.py

```diff
@@ -129,21 +135,24 @@ def __init__(self, payload: Coroutine[Any, Any, RT], parent: 'Scope', delay, at)
                     self, err.subject
                 )
                 self._result = None, err.__transcript__
+                self.parent.__child_finished__(self, failed=False)
             except GeneratorExit:
                 # We are NOT allowed to do any async once the generator
                 # exits forcefully.
                 # We should only receive GeneratorExit due to a forceful
                 # termination in self.__close__ or during cleanup.
-                pass
+                self.parent.__child_finished__(self, failed=False)
```

Comment:
What is the idea behind this branch that the child did not fail but no result is assigned.

Reply:
A ``GeneratorExit`` is only thrown when ``payload.close()`` is called. During a simulation, this only happens in ``task.__close__``, which already sets the result. We do not set the result to ``GeneratorExit`` to avoid raising it in other activities.

However, the payload may *also* get closed when the Task is or payload are garbage collected. In this case, no-one has set our result *and* no one can look at it. This means there are cases where it is perfectly valid that we receive ``GeneratorExit`` without a result.